src/cnc/flavors/aws/lambda-lite/1/provision/partials/collection/lambda.tf.j2

```diff
@@ -0,0 +1,93 @@
+{% for environment in env_collection.active_environments %}
+{% for service in environment.services %}
+{% if service.is_serverless %}
+
+################################################################################
+# Locals Lambda
+################################################################################
+
+locals {
+  {{ service.instance_name }}_lambda_infrastructure_suffix = "{{ service.instance_name }}"
+  {{ service.instance_name }}_lambda_log_group = join("/", ["/aws/lambda", local.{{ service.instance_name }}_lambda_infrastructure_suffix])
+}
+
+################################################################################
+# IAM Role | Role Policy for Lambda to access DynamoDB and CloudWatch
+################################################################################
+resource "aws_iam_role" "{{ service.instance_name }}" {
+  name = join("-", [local.{{ service.instance_name }}_lambda_infrastructure_suffix, "lambda"])
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [{
+      Action    = "sts:AssumeRole",
+      Effect    = "Allow",
+      Principal = {
+        Service = "lambda.amazonaws.com"
+      }
+    }]
+  })
+}
+
+resource "aws_iam_policy" "{{ service.instance_name }}" {
+  name        = join("-", [local.{{ service.instance_name }}_lambda_infrastructure_suffix, "lambda"])
+  description = "IAM policy for Lambda to access specified AWS services"
+
+  policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      for resource in var.{{ service.instance_name }}_aws_resources_permission_lambda : {
+        Action = [
+          "${resource}:*"
+        ],
+        Effect   = "Allow",
+        Resource = "*"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "{{ service.instance_name }}" {
+  role       = aws_iam_role.{{ service.instance_name }}.name
+  policy_arn = aws_iam_policy.{{ service.instance_name }}.arn
+}
+
+################################################################################
+# Lambda Function
+################################################################################
+resource "aws_lambda_function" "{{ service.instance_name }}" {
```

Comment:
We need to add an output for the lambda url and export as the service url in the python code. See the run lite flavor for an example of both.

Reply:
https://github.com/coherenceplatform/cnc/blob/main/src/cnc/models/config/service.py#L368

and

https://github.com/coherenceplatform/cnc/blob/main/src/cnc/flavors/gcp/shared/provision/partials/service/cloud_run_service.tf.j2#L36